manager/src/main/java/com/redhat/service/smartevents/manager/workers/resources/ConnectorWorker.java

```diff
@@ -90,6 +91,16 @@ public ConnectorEntity createDependencies(Work work, ConnectorEntity connectorEn
             return connectorEntity;
         }
         if (status.getState() == ConnectorState.READY) {
+            // If the Connector is ready but differs to that required we need to patch it.
+            JsonNode updatedConnectorDefinition = connectorEntity.getDefinition();
+            if (connectorEntity.getGeneration() > 0) {
+                LOGGER.debug("Managed Connector for '{}' [{}] was found but with a different definition. Patching definition.",
+                        connectorEntity.getName(),
+                        connectorEntity.getId());
+                connectorsApi.updateConnector(connector.getId(), updatedConnectorDefinition);
```

Comment:
@manstis For some reason this call doesn't actually patch the connector, not sure why. Have you tried it by yourself?

Reply:
That'd explain things then.